features/google/src/model/importers/utils/lang-constants.js

```diff
@@ -1,23 +1,29 @@
 export const langConstants = {
     AccessLogImporter: {
-        regex: /\/(Access Log Activity|Zugriffsprotokollaktivitäten|Aktiviteter i adgangsloggen)\/(Activities [-–] A list of Google services accessed by|Aktivitäten [-–] eine Liste der Google-Dienste, auf d|Aktiviteter [-–] en liste over Google-tjenester, som d)\.csv$/u,
+        regex: /\/(Access Log Activity|Zugriffsprotokollaktivitäten|Aktiviteter i adgangsloggen)\/(Activities [-–] A list of Google services accessed by|Aktivitäten [-–] eine Liste der Google-Dienste, auf d|Aktiviteter [-–] en liste over Google-tjenester, som d)\.csv$/,
     },
     ActivitiesHtmlImporter: {
-        regex: /\/(My Activity|Meine Aktivitäten|Mine aktiviteter|MyActivity|MeineAktivitäten|MineAktiviteter)\/.*\.html$/u,
+        regex: /\/(My Activity|Meine Aktivitäten|Mine aktiviteter|MyActivity|MeineAktivitäten|MineAktiviteter)\/.*\.html$/,
     },
     ActivitiesJsonImporter: {
-        regex: /\/(My Activity|Meine Aktivitäten|Mine aktiviteter|MyActivity|MeineAktivitäten|MineAktiviteter)\/.*\.json$/u,
+        regex: /\/(My Activity|Meine Aktivitäten|Mine aktiviteter|MyActivity|MeineAktivitäten|MineAktiviteter)\/.*\.json$/,
     },
     SemanticLocationsImporter: {
-        regex: /\/[^/]+\/(Semantic Location History)\/\d+\/[^.]+\.json$/u,
+        regex: /\/[^/]+\/(Semantic Location History)\/\d+\/[^.]+\.json$/,
     },
 };
-export function matchRegex(path, importer) {
+export function matchRegex(path, importerName) {
     // There can be non standard spaces in the path. Replace them with a normal space.
     const sanitizedPath = path.replace(
         /[\u00a0\u1680\u2000-\u200b\u202f\u205f\u3000\ufeff]/g,
         " "
     );
-    const importerName = importer.constructor.name;
-    return langConstants[importerName].regex.test(sanitizedPath);
+    const normalizedPath = sanitizedPath.normalize("NFC");
+    const normalizedRegex = RegExp(
+        langConstants[importerName].regex
+            .toString()
+            .normalize("NFC")
+            .slice(1, -1)
+    );
+    return normalizedRegex.test(normalizedPath);
```

Comment:
So, I'm all in for normalisation, BUT: I don't understand yet why we need it. The characters we use are all in the two byte range from what I can see, so there shouldn't be any surrogate pairs, hm?

Reply:
They can be non-normalized in some cases, as far as I can tell. That's what happened with filenames.

Reply:
The problem is that you can have more than one way of representing the same character. I was expecting unicode regexes to take care of that but apparently not. 

As far as I understand what unicode regexes do is that they can recognise the unicode codes for commonly used characters like `\u0009 — Tab — \t`. So instead of `\t` you can put `\u0009` in the string and it will still match it. They don't seem to handle normalization at all.

Reply:
Yeah, I did a little more research, and that's also what I ended up with. Fascinatingly, you can even represent a character that has a 2 or even 1 byte codepoint as "surrogate pair" - so yeah, we totally need normalisation on strings coming from weird places even if they only use ASCII characters :D